File: Algorithms/Searching/linearSearch.js
/* Linear Search implementation in JavaScript
 * O(n)
 */

function LinearSearch(elements, x){
	// Linear Search performed on elements to found x
	for (var i=0; i < elements.length; i++){
		if (elements[i] === x){
			return i;
		}
	}
	return -1
}

let elements = [3, 1, 2, 4, 6, 7, 8, 49, 33, 192]
let x = 4; // element to be found


console.log("The elemtents are ", elements)
console.log("The element to be found is: ", x)

let position = LinearSearch(elements, x)
if (position === -1){
	console.log("The elements has not been found")
} else{
	console.log("The element has been found at position ", position + 1)
}

File: Algorithms/Searching/linearSearch.R
# Linear Search R implementation. O(n).
LinearSearch <- function (elements, x){
	# elements : vector with the elements
	# x        : element to be found
	if (length(elements)!= 0){
		for (i in 1:length(elements)){
			if (elements[i] == x){
				return(i)
			}
		}
	}
	return(-1)
}

elements <- c(1, 3, 4, 5, 6, 8, 23, 43, 44, 45)
x <- 23

position <- LinearSearch(elements, x)

if (position[1] == -1){
	print("The element has not been found")
} else {
	sprintf("The element has been fount at position %d", position + 1)
}

